app/services/clients/adminusers.client.js

```diff
@@ -32,17 +33,12 @@ module.exports = function (clientOptions = {}) {
    * @param {string} externalId
    * @return {Promise<User>} A promise of a User
    */
-  function getUserByExternalId (externalId) {
-    return baseClient.get(
-      {
-        baseUrl,
-        url: `${userResource}/${externalId}`,
-        json: true,
-        description: 'find a user',
-        service: SERVICE_NAME,
-        transform: responseBodyToUserTransformer
-      }
-    )
+  async function getUserByExternalId (externalId) {
+    const url = `${baseUrl}${userResource}/${externalId}`
```

Comment:
Ideally this should use `encodeURIComponent` or a URL builder to avoid injection vulnerabilities or encoding issues if special chars are included.

Reply:
Good point. Presumably this should be done for for all constructs of url with query string parameters (or at least those though to be provided by user input)?
@iqbalgds  and I thought this may be better done as a separate PR.

Reply:
So I tend to do it with everything that's not hard-coded but it's particularly important for user input.  In some projects I end up creating a URL model that would be something like:

```
const url = buildUrl({
  base: baseUrl,
  uriParts: [userResource, externalId],
  query: {
    theKey: 'theValue'
  }
})
```

For this ticket I think it's reasonable to just use `encodeURIComponent` on each piece of user input in URLs, but personally if it were me doing it I'd rather use it for all URL parts that aren't hard-coded as  I'd find that to be less effort than figuring out which ones are or might be from user input.